echion/frame.h

```diff
@@ -459,14 +463,31 @@ Frame &Frame::read_local(_PyInterpreterFrame *frame_addr, PyObject **prev_addr)
     // https://github.com/python/cpython/issues/100987#issuecomment-1485556487
     PyObject f_executable;
 
-    for (;frame_addr; frame_addr = frame_addr->previous) {
+    for (;;) {
         // TODO: Cache the executable address for faster reads.
         if (copy_type(frame_addr->f_executable, f_executable)) {
             throw Frame::Error();
         }
         if (f_executable.ob_type == &PyCode_Type) {
             break;
         }
+
+        // Get the previous frame
+        auto prev_frame_addr = frame_addr->previous;
+        if (prev_frame_addr == NULL) {
+            break;
+        }
+
+        // There is a previous frame, so we need to resolve it.
+        frame_addr = stack_chunk->resolve(prev_frame_addr);
+        if (frame_addr == prev_frame_addr) {
+            // The frame is not within the stack chunk so we need to copy it
+            _PyInterpreterFrame iframe;
```

Comment:
Don't you need to put this before the for loop? As you're taking address of this stack allocated variable and using it out of this if block, and the object is stack allocated within this if block.

Reply:
I believe the stack allocation happens on frame entry, so it shouldn't matter where we put `_PyInterpreterFrame iframe;` 🤔